returnn/tf/layers/basic.py

```diff
@@ -2759,45 +2759,89 @@ class SplitDimsLayer(_ConcatInputLayer):
   i.e. the input is (batch, time, window * feature),
   you can set axis="F", dims=(window, -1),
   and you will get the output (batch, time, window, feature).
+
+  If the split axis has a dynamic length,
+  exactly one of the axes that we split into need to also have a dynamic length.
+  You can e.g. use this to split the input dimension into smaller "chunks" of a fixed window size.
+  E.g. you could have input (batch, time, feature) and set axis="T", dims=(-1, window),
+  to get output (batch, split_time, window, feature).
+  In this case, the exact sequence lengths are lost and everything is padded to multiples of the window size using
+  the given padding value.
+  Use :class:`ReinterpretDataLayer` to receive back the original sequence lengths after merging.
+
   Also see :class:`SplitBatchTimeLayer`.
+  Also see :class:`MergeDimsLayer` which can undo this operation.
   """
   layer_class = "split_dims"
 
-  def __init__(self, axis, dims, **kwargs):
+  def __init__(self, axis, dims, pad_to_multiples=None, pad_value=0, **kwargs):
     """
     :param str axis: e.g. "F"
     :param tuple[int]|list[int] dims: what the axis should be split into. e.g. (window, -1)
+    :param Optional[bool] pad_to_multiples: If true, will pad along the axis to the next multiple of the remaining dims.
+      By default this is done iff the axis has a dynamic size
+    :param int|float pad_value: What pad value to use for pad_to_multiples
     """
     super(SplitDimsLayer, self).__init__(**kwargs)
     data = self.input_data
     if isinstance(axis, int):
       data = data.copy_as_batch_major()
+    axis = data.get_axis_from_description(axis)
+    if pad_to_multiples is None:
+      pad_to_multiples = data.is_axis_dynamic(axis)
+
     from returnn.tf.util.basic import get_shape
     old_shape = get_shape(data.placeholder)
-    axis = data.get_axis_from_description(axis)
     new_shape = old_shape[:axis] + list(dims) + old_shape[axis + 1:]
     assert len(new_shape) == len(self.output.batch_shape)
     for i in range(len(new_shape)):
       if new_shape[i] == -1 and self.output.batch_shape[i] is not None:
         new_shape[i] = self.output.batch_shape[i]
+
+    import numpy
+    new_pos_dims = [d for d in dims if d > 0]
+    constant_size = int(numpy.prod(new_pos_dims))
+    assert not data.is_axis_dynamic(axis) or pad_to_multiples or constant_size == 1
+    if pad_to_multiples and constant_size != 1:
+      assert len([d for d in dims if d == -1]) == 1
+      old_size = old_shape[axis]
+      pad_size = (-old_size) % constant_size
+
+      paddings = [(0, 0)] * axis + [(0, pad_size)] + [(0, 0)] * (data.batch_ndim - axis - 1)
+      data.placeholder = tf.pad(data.placeholder, paddings=paddings, constant_values=pad_value)
+
+      axis_wo_batch = data.get_batch_axis_excluding_batch(axis + dims.index(-1))
+      dyn_size = (old_size + pad_size) // constant_size
+      self.output.size_placeholder[axis_wo_batch] = dyn_size
```

Comment:
This is wrong in general, right?
You should use the old `size_placerholder` instead of `old_size`, right?
Also, I thought this code is also for the case that it's not of dynamic length? (Or if not supported, just check and throw `NotImplementedError` otherwise.)

Reply:
Ah yes, this wrong. I will have to fix that, and use `size_placeholder` instead (if it exists, else just the static dim)